src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -254,61 +248,63 @@ def _rgw_add(self, svc_arg=None, inbuf=None):
                 msg = 'Failed to read JSON input: {}'.format(str(e)) + usage
                 return HandleCommandResult(-errno.EINVAL, stderr=msg)
         elif svc_arg:
-                rgw_spec = orchestrator.RGWSpec()
-                rgw_spec.zone_name = svc_arg
-
-        spec = orchestrator.StatelessServiceSpec()
-        spec.service_spec = rgw_spec
-        spec.name = rgw_spec.rgw_zone
+            rgw_spec = orchestrator.RGWSpec()
+            rgw_spec.zone_name = svc_arg
+        else:
+            return HandleCommandResult(-errno.EINVAL, stderr=usage)
 
-        return self._add_stateless_svc("rgw", spec)
+        completion = self.add_rgw(rgw_spec)
+        self._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        return HandleCommandResult(stdout=completion.result)
```

Comment:
Given the "happy path" always results in these same 3 lines, any chance to replace with a method?
```py
return WrapCommand(self.add_rgw(rgw_spec))
```

Reply:
#26565